share/rocm/cmake/ROCMSetupVersion.cmake

```diff
@@ -152,9 +152,33 @@ function(rocm_setup_version)
 
 endfunction()
 
+function(rocm_get_so_patch OUTPUT)
+    set(_patch "0")
+    if(DEFINED ENV{ROCM_LIBPATCH_VERSION})
+        set(_patch $ENV{ROCM_LIBPATCH_VERSION})
+    endif()
+    rocm_set_parent(${OUTPUT} ${_patch})
+endfunction()
+
 function(rocm_set_soversion LIBRARY_TARGET SOVERSION)
     if(NOT WIN32)
-        rocm_version_regex_parse("^([0-9]+).*" SOVERSION_MAJOR "${SOVERSION}")
-        set_target_properties(${LIBRARY_TARGET} PROPERTIES VERSION ${SOVERSION} SOVERSION ${SOVERSION_MAJOR})
+        rocm_version_regex_parse("^([0-9]+).*" LIB_VERSION_MAJOR "${SOVERSION}")
+        rocm_version_regex_parse("^[0-9]+\\.(.*)" LIB_VERSION_MINOR "${SOVERSION}")
+
+        rocm_get_so_patch(LIB_VERSION_PATCH)
+        set (LIB_VERSION_STRING "${LIB_VERSION_MAJOR}.${LIB_VERSION_MINOR}.${LIB_VERSION_PATCH}")
+
+        set_target_properties(${LIBRARY_TARGET} PROPERTIES SOVERSION ${LIB_VERSION_MAJOR} )
+        set_target_properties(${LIBRARY_TARGET} PROPERTIES VERSION ${LIB_VERSION_STRING} )
+    endif()
+endfunction()
+
+function(rocm_set_runpath)
+    if(NOT WIN32 AND NOT APPLE)
+        if(${ARGC} GREATER 0)
+            set ( CMAKE_SHARED_LINKER_FLAGS " -Wl,--enable-new-dtags -Wl,--rpath,${ARGV0}" PARENT_SCOPE )
```

Comment:
These should be set by the toolchain not by rocm-cmake. We should never be overriding these variables.

Reply:
Sorry I do get what you mean by toolchain. Is it that we should never have a api in rocm cmake to set the runpath? Can you please clarify.

Reply:
> Is it that we should never have a api in rocm cmake to set the runpath?

More than that, it should never appear in the CMakeLists.txt ever. Instead this should be passed in when running cmake either by passing `-DCMAKE_SHARED_LINKER_FLAGS=...` or putting it in a cmake toolchain file and passing the file in with `-DCMAKE_TOOLCHAIN_FILE=...`.

Reply:
Also, if you are using `rbuild` the `-D` variables can be passed to it as well or you can pass the cmake toolchain file with `--toolchain` flag.

Reply:
I understand. I will remove this function (rocm_set_runpatch) and 2 test cases which check this api.